Cleanup

IEvangelist opened the pull request:
Cleanup:

- Split classes into separate files
- Fixed logging perf issues, avoided interpolated messages, and instead used format.
- Consistent file-scoped namespaces.
- Use `GlobalUsings` where it makes sense.
- Use `readonly record struct` instead of `record` (or `record class`) where possible.
- Delete redundant _CODE_OF_CONDUCT(2).md_ file.
- Make classes `sealed` where applicable.
- Use pattern matching where it makes sense for readability and succinctness.
- Consistent naming of request objects.
- Use HTTPS alternative URLs for RSS feeds when seeding DB data, where possible.
- A bit more CSS for the recently added playback rate control.

migueBarrera commented:
@IEvangelist  could you do this pr into release/next-steps branch instead of main branch.
@jamesmontemagno  what do you think?

jamesmontemagno commented:
@migueBarrera switched but lots of conflicts. Any reason not to just merge the next-steps stuff into main? nothing really breaking correct?

IEvangelist commented on src/MobileBlazor/mauiapp/MauiProgram.cs:
```diff
@@ -11,7 +11,7 @@ public static class MauiProgram
     public static string BaseWeb = $"{Base}:5002/listentogether";
     public static string Base = DeviceInfo.Platform == DevicePlatform.Android ? "http://10.0.2.2" : "http://localhost";
-    public static string APIUrl = $"{Base}:5000/v1/";
+    public static string APIUrl = $"{Base}:5000/";
```
```suggestion
    public static string APIUrl = $"{Base}:5000/v1/";
```

migueBarrera commented:
next-steps has a lot of changes from the last few weeks.. 
@IEvangelist  made his changes in main instead of next-steps. The other PR from daniel it was me who resolved the conflicts with next-steps. I can do it with this PR again, I know all the changes. 
Also, I can merge next-steps over main and we avoid these problems again.

migueBarrera commented:
the next-steps branch has been merged into main.
@IEvangelist you need resolve conflicts. I can't do it, it's in a repo of yours.

IEvangelist commented:
> the next-steps branch has been merged into main. @IEvangelist you need resolve conflicts. I can't do it, it's in a repo of yours.

Done...

IEvangelist commented:
Hey @jamesmontemagno - are we good to merge this one?

jamesmontemagno commented:
@IEvangelist just changed to merge to main, can you re-validate stuff for me.

IEvangelist commented:
> @IEvangelist just changed to merge to main, can you re-validate stuff for me.

Hey @jamesmontemagno - done

IEvangelist commented:
I'm curious, why was this closed? I've been doing everything that you've asked of me on this one. :octocat: 

I've already contributed several features to this app and would love to know what happened here.
- I've previously [added the playback rate controls](https://github.com/microsoft/dotnet-podcasts/commit/41d682d3e14014ed8e6523dbf12f845a0922e9ef) that were originally missing.
- I've also provided an initial effort, which was [lots of C# and Blazor clean-up, and I added a system theme](https://github.com/microsoft/dotnet-podcasts/commit/245b665b3d76d92a6422f3bf4860ba15e3345854) support.

[This PR adds a ton of value](https://github.com/microsoft/dotnet-podcasts/pull/122#issue-1233090234) and addressed a lot of inconsistencies with how some of this stack is now documented. ☹️

lockhartsoftware commented:
Would someone please progress this PR? Thanks.
